File: my_solutions/1724.checking-existence-of-edge-length-limited-paths-ii/solution.cpp
#include <vector>
#include <algorithm>
#include <climits>
using namespace std;

class PersistentUnionFind {
private:
    vector<int> rank;
    vector<int> parent;
    vector<int> version;

public:
    PersistentUnionFind(int n)
        : rank(n, 0)
        , parent(n)
        , version(n, INT_MAX) {
            // TODO: implement
        }
};

File: my_solutions/1724.checking-existence-of-edge-length-limited-paths-ii/solution.java
class PersistentUnionFind {
    private final int inf = 1 << 30;
    private int[] rank;
    private int[] parent;
    private int[] version;

    public PersistentUnionFind(int n) {
        rank = new int[n];
        parent = new int[n];
        version = new int[n];
        for (int i = 0; i < n; i++) {
            parent[i] = i;
            version[i] = inf;
        }
    }

    public int find(int x, int t) {
        // TODO: implement
    }
}
